examples/copas-example.lua

```diff
@@ -16,31 +15,39 @@ local ping = mqtt.client{
 	username = token,
 	clean = true,
 	version = mqtt.v50,
+	-- NOTE: copas connector
+	connector = require("mqtt.luasocket-copas"),
 }
 
 local pong = mqtt.client{
 	uri = "mqtt.flespi.io",
 	username = token,
 	clean = true,
 	version = mqtt.v50,
+	-- NOTE: copas connector
+	connector = require("mqtt.luasocket-copas"),
 }
 
 ping:on{
 	connect = function(connack)
 		assert(connack.rc == 0)
 		print("ping connected")
 
-		for i = 1, num_pings do
-			copas.sleep(timeout)
-			print("ping", i)
-			assert(ping:publish{ topic = "luamqtt/copas-ping/"..suffix, payload = "ping"..i, qos = 1 })
-		end
-
-		copas.sleep(timeout)
-
-		print("ping done")
-		assert(ping:publish{ topic = "luamqtt/copas-ping/"..suffix, payload = "done", qos = 1 })
-		ping:disconnect()
+		-- adding another thread; copas handlers should return quickly, anything
+		-- that can wait should be off-loaded from the handler to a thread.
+		-- Especially anything that yields; socket reads/writes and sleeps, and the
+		-- code below does both, sleeping, and writing (implicit in 'publish')
+		copas.addthread(function()
+			for i = 1, num_pings do
+				copas.sleep(delay)
+				print("ping", i)
+				assert(ping:publish{ topic = "luamqtt/copas-ping/"..suffix, payload = "ping"..i, qos = 1 })
+			end
+
+			print("ping done")
+			assert(ping:publish{ topic = "luamqtt/copas-ping/"..suffix, payload = "done", qos = 1 })
+			ping:disconnect()
+		end)
```

Comment:
@xHasKx fyi, these examples might have accidentally worked, but were essentially broken in a bad way.